sklearn/preprocessing/_data.py

```diff
@@ -3428,9 +3428,18 @@ def inverse_transform(self, X):
             "yeo-johnson": self._yeo_johnson_inverse_transform,
         }[self.method]
         for i, lmbda in enumerate(self.lambdas_):
-            with np.errstate(invalid="ignore"):  # hide NaN warnings
-                X[:, i] = inv_fun(X[:, i], lmbda)
-
+            with warnings.catch_warnings(record=True) as captured_warnings:
+                with np.errstate(invalid="warn"):
+                    X[:, i] = inv_fun(X[:, i], lmbda)
+            if captured_warnings and np.isnan(X[:, i]).any():
```

Comment:
Running `np.isnan(X[:, i])` creates a NumPy array of shape `(n_sample, )`, which adds memory overhead.

I know it's more brittle, but can we check that the NumPy warning message contains "invalid value encountered in power" and then raise our own warning?

Reply:
I created a warning called TransformationFailedWarning and used it for a more straightforward check, as you suggested.

Reply:
Sorry for being unclear. I mostly wanted to convert `np.isnan(X[:, i]).any()` to `"invalid value encountered in power" in warning_message` and leave everything else the same.

Can you revert the new warning object?

Reply:
Done.